File: Solutions/1221. Split a String in Balanced Strings/1221. Split a String in Balanced Strings.go
func balancedStringSplit(s string) int {
    res,lcnt,rcnt:=0,0,0
    for _,c:=range(s){
        if c=='R'{
            rcnt+=1
        }else{
            lcnt+=1
        }
        if rcnt!=0 && rcnt==lcnt{
            res+=1
        }
    }
    return res
}

func balancedStringSplit(s string) int {
    res,cnt:=0,0
    for _,c:=range(s){
        if c=='L'{
            cnt+=1
        }else{
            cnt-=1
        }
        if cnt==0{
            res+=1
        }
    }
    return res
}

File: Solutions/1221. Split a String in Balanced Strings/1221. Split a String in Balanced Strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int res=0,lcnt=0,rcnt=0;
        for(var c:s.toCharArray()){
            if(c=='R')
                ++rcnt;
            else if(c=='L')
                ++lcnt;
            if(rcnt!=0 && rcnt==lcnt)
                ++res;
        }
        return res;
    }
}

class Solution {
    public int balancedStringSplit(String s) {
        int res=0,cnt=0;
        for(var c:s.toCharArray()){
            cnt+=c=='L'?1:-1;
            if(cnt==0)++res;
        }
        return res;
    }
}